src/app/components/scan-fingers/scan-fingers.component.ts

```diff
@@ -1,20 +1,88 @@
-import { Component, OnInit } from '@angular/core';
-import { FingerprintReader } from '@digitalpersona/devices';
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import {
+  FingerprintReader,
+  QualityCode,
+  QualityReported,
+  SampleFormat,
+} from '@digitalpersona/devices';
+
+enum ScanStatus{
+  Successful= 'Successful',
+  Failed = 'Failed'
+}
 
 @Component({
   selector: 'app-scan-fingers',
   templateUrl: './scan-fingers.component.html',
-  styleUrls: ['./scan-fingers.component.scss']
+  styleUrls: ['./scan-fingers.component.scss'],
 })
-export class ScanFingersComponent implements OnInit {
 
+export class ScanFingersComponent implements OnInit {
   private reader!: FingerprintReader;
 
-  constructor() { }
+  @Input() isReaderConnected: boolean;
+  @Output() isReaderConnectedChange = new EventEmitter<boolean>();
+
+  @Input() step: number;
+  @Output() stepChange = new EventEmitter<number>();
+
+  scanStatus?: ScanStatus;
+
+  constructor() {
+    this.isReaderConnected = false;
+    this.step = 1;
+    this.scanStatus = undefined;
+  }
 
   ngOnInit(): void {
     this.reader = new FingerprintReader();
-    this.reader.enumerateDevices().then(console.log)
+    this.reader.on('DeviceConnected', () => {
+      this.handleDeviceConnectionStatusChange(true);
+    });
+    this.reader.on('DeviceDisconnected', (event) => {
+      this.handleDeviceConnectionStatusChange(false);
+    });
+    this.reader.on('QualityReported', (event: QualityReported) => {
+      const { quality } = event;
+      this.handleQualityReport(quality);
+    });
+    this.reader.on('SamplesAcquired', async (event) => {
+      console.log({ event });
+      await this.reader.stopAcquisition();
+      this.reader
+        .startAcquisition(SampleFormat.Intermediate)
+        .then((eventStart) => {
+          console.log({ eventStart });
+        });
+    });
+    this.reader.onAcquisitionStarted = (event) => {
+      console.log({ event });
+    };
+    this.reader.onAcquisitionStopped = (eventStop) => {
+      console.log({ eventStop });
```

Comment:
remove all events/functions that we don't use at the moment

Reply:
@hoangngk Removed